x-pack/legacy/plugins/alerting/server/types.ts

```diff
@@ -8,6 +8,9 @@ import { AlertInstance } from './alert_instance';
 import { AlertTypeRegistry as OrigAlertTypeRegistry } from './alert_type_registry';
 import { PluginSetupContract, PluginStartContract } from './plugin';
 import { SavedObjectAttributes, SavedObjectsClientContract } from '../../../../../src/core/server';
+import { Alert } from '../common';
+
+export * from '../common';
 
```

Comment:
This pattern keeps just one touch point between `server` and `common`, the rest of `server` should use this module to consume its types.

This helps address one annoying issue in how we use TS that it forces us to change dozens of places whenever types are changed/moved, making it a more painful process than it needs to be.
It also helps reduce coupling by reducing the touch points between modules.